client/tx/encode.go

```diff
@@ -19,7 +19,8 @@ import (
 type (
 	// EncodeReq defines a tx encoding request.
 	EncodeReq struct {
```

Comment:
@yangyanqing what I meant was remove the `EncodeReq` entirely. So we have the following:

```go
func EncodeTxRequestHandlerFn(cdc *codec.Codec, cliCtx context.CLIContext) http.HandlerFunc {
	return func(w http.ResponseWriter, r *http.Request) {
		var req auth.StdTx

		body, err := ioutil.ReadAll(r.Body)
		if err != nil {
			rest.WriteErrorResponse(w, http.StatusBadRequest, err.Error())
			return
		}

		err = cdc.UnmarshalJSON(body, &req)
		if err != nil {
			rest.WriteErrorResponse(w, http.StatusBadRequest, err.Error())
			return
		}

		// re-encode it via the Amino wire protocol
		txBytes, err := cliCtx.Codec.MarshalBinaryLengthPrefixed(req)
		if err != nil {
			rest.WriteErrorResponse(w, http.StatusInternalServerError, err.Error())
			return
		}

		// base64 encode the encoded tx bytes
		txBytesBase64 := base64.StdEncoding.EncodeToString(txBytes)

		response := EncodeResp{Tx: txBytesBase64}
		rest.PostProcessResponse(w, cdc, response, cliCtx.Indent)
	}
}
```

Reply:
Ref 2 and 3 in https://github.com/cosmos/cosmos-sdk/issues/4141#issuecomment-484312785
I can not find a way satisfied both single endpoint and single swagger-ui.
So, this PR was WIP. More discussion was needed in #4141.

Reply:
I would argue the endpoint should be simple. Swagger limitations should not limit us from writing proper endpoint implementations.

Reply:
In addition to swagger, are there other documentation describing this interface?
Wrong documents will cause very big trouble !